src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.NativeAot.cs

```diff
@@ -319,6 +319,11 @@ public static void PrepareDelegate(Delegate d)
                 throw new NotSupportedException(SR.NotSupported_ManagedActivation);
             }
```

Comment:
(Unrelated to this PR)

We are missing a check for function pointers here. Could you please fix it in a follow up?

Reply:
Activator.CreateInstance seems to be missing it too.

Reply:
Not close to a computer now but I think Activator should be fine because there's no constructor to run and they're not a valuetype. I don't see it handling pointers either.